### PR TITLE
Consider hostname specified in env config file

### PR DESCRIPTION
**Pull request by marcandrews**
My VPS has two IPs, and when I start my MEAN app, the server listens on both IPs. This way, the MEAN app takes into consideration the hostname specified in the env config file when starting the server.

### CODE REVIEW

**Review by timelf123 — APPROVED** [review]
lgtm after merges fixed - this conforms with the http server listen api

**Comment by clowenhg** [comment]
My concern is for sites that are using the `hostname` config items for building urls, without the intention of limiting their server to only listening to a specific hostname.

We should update this to have a flag that can be turned on to tell it to use the hostname for the listen call.

**Pull request closed without merging**
